samcli/commands/delete/delete_context.py

```diff
@@ -319,12 +334,14 @@ def delete(self):
             self.cf_utils.delete_stack(stack_name=self.stack_name, retain_resources=retain_resources)
             self.cf_utils.wait_for_delete(self.stack_name)
 
-        # If s3_bucket information is not available, warn the user
-        if not self.s3_bucket:
-            LOG.debug("Cannot delete s3 files as no s3_bucket found")
+        # Warn the user that s3 information is missing and to use --s3 options
+        if not self.s3_bucket or not self.s3_prefix:
+            LOG.debug("Cannot delete s3 objects as bucket or prefix is missing")
```

Comment:
Question mark: is providing s3 bucket without a prefix not a valid use case?

Reply:
Thats a good point! The user could make the choice to not have the prefix be deleted. We could revert the warning message+conditions back or remind the user that `sam delete` did not delete any prefix data.